### ISSUE TITLE
Magento 1.9.* Welcome message problems

### ISSUE THREAD

**Issue by TerfanAybara** [GordonLesti/Lesti_Fpc#311]
This is a very random problem and has weird results.

When a customer logs in their name will show up in the Welcome Message. When they either click log out or their session expires the Welcome message changes to "Welcome, Gordon Lesti! (Not Gordon Lesti?)" This is where things get weird. The first time you visit a product page (or any page that is cache-bale) or refresh the cache in Magento, the message acts as you would expect, but on refreshing the page or when navigating away and back to that product page, the welcome message shows either nothing in the first section or has the default welcome message in the first section, but still has "(Not Gordon Lesti?)". For example it will either be "Default welcome msg! (Not Gordon Lesti?)" or just "(Not Gordon Lesti?)" with no welcome message. I have seen it do both. I have also seen it give the wrong customer name.

I have tried taking the welcome message out of the lazy block and put it in the dynamic block. I have tried adding the parent blocks so they are not getting cached, but nothing seems to help. On pages that are not cached (like the cart page) or if I turn off FPC, the message goes back to normal. 

I have been searching for a solution and have not found one. This problem can be recreated with a fresh install of Magento 1.9.* (I did my install with both 1.9.3.6 and 1.9.3.7 and without the default products, but added my own for testing) and the only module installed is Lesti FPC. In System->Configuration->Customers->Persistent Shopping Cart, turn it on and turn off Clear Persistence on Log Out. All other settings should be default. Please help!!!

**Comment by kvnx** [comment]
yes i have tha same problem an i have welcome message in lazy block. I'cant understand how to solve this!

**Comment by danielcmuresan** [comment]
@TerfanAybara try to add in Dynamic blocks this: "header.additional"
Having "welcome" in Lazy Blocks it's fine but (Not Gordon Lesti?) comes from persistent shopping cart.
I have this issue as well and I added this, so far I haven't seen the problem anymore.
Please let me know if this worked for you as well

**Comment by kvnx** [comment]
I don't understand what is this "(Not Gordon Lesti?)" .... ??

**Comment by danielcmuresan** [comment]
The name is not important, "Gordon Lesti" was added as an example.
![image](https://user-images.githubusercontent.com/609253/38026136-9d2ca5b8-3293-11e8-8b16-eeda8f0a52de.png)

**Comment by kvnx** [comment]
Ahahaan OK thank you. Excuse me! but I must also keep "welcome" in lazy or not? additional do i have to put it lazy or dynamic?

**Comment by danielcmuresan** [comment]
Yes, keep "welcome" in lazy.
"additional" I would say that can go to dynamic.
and make sure you have magento > 1.7

also, keep in mind that this may not work, it's still something that needs to be tested.

**Comment by kvnx** [comment]
OK thank you. I'll try this in this afternoo.There are sale way for test if work?

**Comment by triplexita** [comment]
is this?

**Comment by kvnx** [comment]
Is this ok? i have this in dynamic

messages,
header.additional,
global_messages,
global_notices,
giantpoints.toplinks.label,
global_cookie_notice,
right.reports.product.viewed,
compare.mini

and this in lazy

welcome,
top.links,
giantpoints.toplinks.label,
cart_sidebar,
catalog.compare.sidebar,
wishlist_sidebar,
minicart_head,
right.poll

**Comment by danielcmuresan** [comment]
seems to be right

**Comment by TerfanAybara** [comment]
@danielcmuresan I will test this out and get back to you. Thanks for the help and suggestion

**Comment by tawfekov** [comment]
in case you was using the popular module : google analytics plus , you should add this block : googleanalyticsplus_universal

other blocks can be found on : app/design/frontend/base/default/layout/googleanalyticsplus.xml
